scripts/native_simulator/native/src/include/nsi_cpu0_interrupts.h

```diff
@@ -9,6 +9,7 @@
 #define TIMER_TICK_IRQ 0
 #define OFFLOAD_SW_IRQ 1
 #define COUNTER_EVENT_IRQ 2
+#define NSOS_IRQ 3
```

Comment:
Better define it in https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/posix/native_sim/board_soc.h#L24 
and include soc.h in nsos_adapt.c
(This nsos_adapt code is zephyr specific, while this file is part of the upstream native simulator. Booking interrupts upstream for Zephyr only use is a bit out of place)

Reply:
Moved this definition to `board_soc.h` as suggested. Should it still be value of 3 or is that range reserved for upstream interrupts only?

Reply:
By now we can just let it be a hardcoded 3. Later I can add a macro in the native simulator header keeping the first available interrupt number for the user, and then we can define it based on that.